numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2059,18 +2059,22 @@ array_fromfile(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *keywds)
     PyObject *err_type = NULL, *err_value = NULL, *err_traceback = NULL;
     char *sep = "";
     Py_ssize_t nin = -1;
-    static char *kwlist[] = {"file", "dtype", "count", "sep", NULL};
+    static char *kwlist[] = {"file", "dtype", "count", "sep", "offset", NULL};
     PyArray_Descr *type = NULL;
     int own;
-    npy_off_t orig_pos = 0;
+    npy_off_t orig_pos = 0, offset = 0;
     FILE *fp;
 
     if (!PyArg_ParseTupleAndKeywords(args, keywds,
-                "O|O&" NPY_SSIZE_T_PYFMT "s:fromfile", kwlist,
-                &file, PyArray_DescrConverter, &type, &nin, &sep)) {
+                "O|O&" NPY_SSIZE_T_PYFMT "sl:fromfile", kwlist,
+                &file, PyArray_DescrConverter, &type, &nin, &sep, &offset)) {
```

Comment:
I'm not sure `l` is the right choice - `long int` is too short on windows.

CPython uses the following in `mmapmodule.c`:
```
#ifdef UNIX
#ifdef HAVE_LARGEFILE_SUPPORT
#define _Py_PARSE_OFF_T "L"
#else
#define _Py_PARSE_OFF_T "l"
#endif
```

Reply:
Do you think ```NPY_OFF_T_PYFMT``` from [npy_common.h](https://github.com/numpy/numpy/blob/58da56876328e51473ebd4c5ce819262621d5278/numpy/core/include/numpy/npy_common.h#L157) would suffice?

Reply:
Good find, that's exactly what is needed here.